ipamd/ipamd.go

```diff
@@ -707,9 +707,14 @@ func (c *IPAMContext) waitENIAttached(eni string) (awsutils.ENIMetadata, error)
 	// Wait until the ENI shows up in the instance metadata service
 	retry := 0
 	for {
+		retry++
```

Comment:
This code appears to retry one time more than the max retries.

Reply:
Moved the increase to the end of the loop.